nativelib/src/main/scala/scala/re2s/UnicodeTableData.scala

```diff
@@ -0,0 +1,2572 @@
+package scala.re2s
+
+trait UnicodeTableData {
```

Comment:
I understand & agree.  I think you have read and support my concerns with "Single Points of
Truth"

I will check.  I did a first pass and nothing obvious fell out.

After I get `breakable` resolved and the unit-tests working in Travis CI 
(they work on my machine) I'll try to find my stash of clever and take a more extended 
look.

If something is possible, I will probably write some code to make sure that the
common code matches what is currently in Character.scala and re2s.

Reply:
Given the topic here about reducing binary code size, can I ask an implementation
question.

Normally I would consider scalanative.re2s and its unicode data to be more
primitive that java.lang.Character.scala, so I would have the latter call into the former.

I am guessing that Character.scala gets brought into even the simplest HelloWorld 
program.  I am also guessing that re2s will be brought in relatively infrequently.
Do I need to put the Unicode data in a separate `object` in re2s so that only it
is brought in by Character.scala, not all  or too much of re2s?  

Since correctness & time-available are competing, If Character is always being
brought in, I could reverse things and have re2s, if linked, call into Character. 
A bit weird, but i could comment it.

Reply:
We can make a unicode module somewhere in runtime with all of the unicode data, and then both Character and re2s can both call to it.

Reply:
Understood & agreed. Thank you for the `runtime` placement suggestion.

Reply:
Created Issue #1542 as a result of discussions.